starlette/middleware/servertiming.py

```diff
@@ -0,0 +1,121 @@
+import threading
+import time
+import typing
+
+from starlette.middleware.base import BaseHTTPMiddleware, RequestResponseEndpoint
+from starlette.requests import Request
+from starlette.types import ASGIApp
+
+
+class Ticker:
+    def __init__(self, name="app", description="main app"):
+        self.name = name
+        self.description = description
+        self._start = self._end = None
+
+    def start(self):
+        self._start = time.time()
+        _add_ticker(self)
+
+    def end(self):
+        self._end = time.time()
```

Comment:
Have you considered using `time.monotonic()` instead of `time.time()`?
It still returns fractional seconds but it's slightly safer.